app/controllers/spree/checkout_controller_decorator.rb

```diff
@@ -6,38 +6,73 @@
   private
 
     def add_gift_card_payments
```

Comment:
why we update the functionality of add_gift_card_payment?

Reply:
When a guest user makes a payment through a claim code, this functionality was not available in the gem. It had to be manually handled and updated in this methid

Reply:
When a guest user makes a payment through a claim code, this functionality was not available in the gem. It had to be manually handled and updated in this method